services/encoding_service.go

```diff
@@ -389,6 +389,80 @@ func (s *EncodingService) RetrieveMP4MuxingCustomData(encodingID string, mp4ID s
 	return &r, nil
 }
 
+func (s *EncodingService) AddProgressiveMOVMuxing(encodingID string, a *models.ProgressiveMOVMuxing) (*models.ProgressiveMOVMuxingResponse, error) {
+	b, err := json.Marshal(*a)
+	if err != nil {
+		return nil, err
+	}
+	path := EncodingEndpoint + "/" + encodingID + "/" + "muxings/progressive-mov"
+	o, err := s.RestService.Create(path, b)
+	if err != nil {
+		return nil, err
+	}
+	var r models.ProgressiveMOVMuxingResponse
+	err = json.Unmarshal(o, &r)
+	if err != nil {
+		return nil, err
+	}
+	return &r, nil
+}
+
+func (s *EncodingService) RetrieveProgressiveMOVMuxing(encodingID string, progressiveMOVID string) (*models.ProgressiveMOVMuxingResponse, error) {
+	path := EncodingEndpoint + "/" + encodingID + "/" + "muxings/progressive-mov" + "/" + progressiveMOVID
+	o, err := s.RestService.Retrieve(path)
+	if err != nil {
+		return nil, err
+	}
+	var r models.ProgressiveMOVMuxingResponse
+	err = json.Unmarshal(o, &r)
+	if err != nil {
+		return nil, err
+	}
+	return &r, nil
+}
+
+func (s *EncodingService) DeleteProgressiveMOVMuxing(encodingID string, progressiveMOVID string) (*models.ProgressiveMOVMuxingResponse, error) {
+	path := EncodingEndpoint + "/" + encodingID + "/" + "muxings/progressive-mov" + "/" + progressiveMOVID
+	o, err := s.RestService.Delete(path)
+	if err != nil {
+		return nil, err
+	}
+	var r models.ProgressiveMOVMuxingResponse
+	err = json.Unmarshal(o, &r)
+	if err != nil {
+		return nil, err
+	}
+	return &r, nil
+}
+
+func (s *EncodingService) ListProgressiveMOVMuxing(encodingID string, offset int64, limit int64) (*models.ProgressiveMOVMuxingListResponse, error) {
+	path := EncodingEndpoint + "/" + encodingID + "/" + "muxings/progressive-mov"
+	o, err := s.RestService.List(path, offset, limit)
+	if err != nil {
+		return nil, err
+	}
+	var r models.ProgressiveMOVMuxingListResponse
+	err = json.Unmarshal(o, &r)
+	if err != nil {
+		return nil, err
+	}
+	return &r, nil
+}
+
+func (s *EncodingService) RetrieveProgressiveMOVMuxingCustomData(encodingID string, progressiveWebMID string, offset int64, limit int64) (*models.CustomDataResponse, error) {
+	path := EncodingEndpoint + "/" + encodingID + "/" + "muxings/progressive-mov" + "/" + progressiveWebMID
```

Comment:
`... + "/custom-data"` missing?

Reply:
Good catch

Reply:
Nope never mind, it's handled in the RetrieveCustomData call on Rest Service

Reply:
cool :+1: